framework/src/org/checkerframework/framework/type/DefaultTypeHierarchy.java

```diff
@@ -1031,6 +1031,10 @@ protected boolean visitWildcardSubtype(
                 // TODO: wildcard capture conversion
                 // Two unannotated uses of reference-equal wildcard types are the same type
                 return true;
+            } else {
+                // Otherwise just check that the upper bounds are in the correct relationship
+                AnnotatedWildcardType superWildcard = (AnnotatedWildcardType) supertype;
+                return isSubtype(subtype.getExtendsBound(), superWildcard.getExtendsBound());
```

Comment:
Why is this the correct relationship? I would have expected `isSubtype(subtype.getExtendsBound(), superWildcard.getSuperBound())`, which would ensure that all possible subtype instantiations are below the lower bound of the supertype. Like this, the subtype could be instantiated with something that is a supertype of the superWildcard, which would seem unsound.